modin/backends/pandas/query_compiler.py

```diff
@@ -3023,3 +3023,12 @@ def cat_codes(self):
         return self.default_to_pandas(lambda df: df[df.columns[0]].cat.codes)
 
     # END Cat operations
+
+    def compare(self, other, **kwargs):
```

Comment:
I'm sorry for the late question but still did you check the performance for moving `other` object to pandas and calling `_apply_full_axis`? Because for some cases this works faster.

Reply:
No problem. Can you please specify what you mean by moving other to pandas? If we convert other to pandas right away using other.to_pandas() then we would get a single pandas dataframe object. We want to perform the compare across likewise partitions between self and other.

Reply:
Broadcast join is probably what @YarShev means. There's no need to implement only for `compare`, we need some cost analysis to do this for all of the `BinaryFunction` operations.

Reply:
Probably it belongs in a separate issue.

Reply:
Yes, it is that what I mean. That makes sense to consider in a separate issue then.